src/liboslexec/closure.cpp

```diff
@@ -76,7 +76,7 @@ print_component_value(std::ostream &out, ShadingSystemImpl *ss,
     else if (type == TypeDesc::TypeVector)
         out << "(" << ((Vec3 *)data)->x << ", " << ((Vec3 *)data)->y << ", " << ((Vec3 *)data)->z << ")";
     else if (type == TypeDesc::TypeString)
-        out << "\"" << ((ustring *)data)->c_str() << "\"";
+        out << "\"" << ((ustring *)data)->string() << "\"";
```

Comment:
`c_str()` can return `NULL` (which happens to be the default value for any string keywords). Outputting `NULL` is documented as undefined behavior for c++ streams. Since this is not performance critical code, the string version seemed safer.